srcCpp/RTIDDSImpl.cxx

```diff
@@ -671,37 +683,35 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
                 _useUnbounded = MAX_BOUNDED_SEQ_SIZE;
             }
             _isLargeData = true;
-        } else if (_scan_max_size <= (unsigned long)(std::min)(MAX_SYNCHRONOUS_SIZE,MAX_BOUNDED_SEQ_SIZE)) {
+        } else {
             _useUnbounded = 0;
             _isLargeData = false;
-        } else {
-            return false;
         }
-        if (_isLargeData && _BatchSize > 0) {
-            fprintf(stderr, "Batching cannnot be used with asynchronous writing.\n");
-            return false;
-        }
-    } else { // If not Scan, compare sizes of Batching and dataLen
-        /*
-         * We don't want to use batching if the sample is the same size as the batch
-         * nor if the sample is bigger (in this case we avoid the checking in the
-         * middleware).
-         */
-        if (_BatchSize > 0 && (unsigned long)_BatchSize <= _DataLen) {
+        // If not Scan, compare sizes of Batching and dataLen
+    } else if (_BatchSize > 0 && (unsigned long)_BatchSize < _DataLen * 2) {
+            /*
+            * We don't want to use batching if the batch size is not large enough
+            * to contain at least two sample (in this case we avoid the checking in
+            * the middleware).
+            */
             fprintf(stderr,
-                    "Batching disabled: BatchSize (%d) is equal or smaller "
-                    "than the sample size (%lu).\n",
+                    "Batching disabled: BatchSize (%d) is smaller than two "
+                    "times the sample size (%lu).\n",
                     _BatchSize,
                     _DataLen);
             _BatchSize = 0;
-        }
     }
 
     if (_DataLen > (unsigned long)MAX_SYNCHRONOUS_SIZE) {
         fprintf(stderr, "Large data settings enabled.\n");
         _isLargeData = true;
     }
 
+    if (_isLargeData && _BatchSize > 0) {
+            fprintf(stderr, "Batching cannot be used with asynchronous writing.\n");
+            return false;
+    }
+
```

Comment:
I just change the order of the if.
This already worked but it makes more sense with this order